src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeption.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationException("User is null");
+        }
+        if (user.getLogin() == null) {
+            throw new ValidationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LENGTH) {
+            throw new ValidationException("Login length is less than 6 characters");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("User with such login exists");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE) {
+            throw new ValidationException("Age of user is less than 18 years");
+        }
+        if (user.getPassword() == null) {
+            throw new ValidationException("Password can't be null");
+        }
+        if (user.getPassword().length() < MIN_LENGTH) {
+            throw new ValidationException("Password length is less than 6 characters");
+        }
```

Comment:
dont forget about checking if user exists with same login

Reply:
https://github.com/mate-academy/jv-registration-validation/pull/944/files#diff-353b797add663a2e7fa812461722f0adf75fe98fbb4d21a1b44501febed948c6R24-R25
Here

Reply:
Test is here
https://github.com/mate-academy/jv-registration-validation/pull/944/files#diff-3ea4f7b55610599aa400ab352db85891e9fd882744226d23ff52065a55f673b1R24-R39